src/Sarif.Viewer.VisualStudio/Models/SarifErrorListItem.cs

```diff
@@ -597,112 +602,100 @@ internal void RemapFilePath(string originalPath, string remappedPath)
         /// <summary>
         /// Attaches to the document using the specified properties, which are also cached.
         /// </summary>
-        internal void AttachToDocument(string documentName, long docCookie, IVsWindowFrame windowFrame)
+        internal bool TryAttachToDocument(string documentName, long docCookie, IVsWindowFrame windowFrame)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (_documentCookie.HasValue || string.Compare(documentName, this.FileName, StringComparison.OrdinalIgnoreCase) != 0)
+            {
+                return false;
+            }
+
             // Cache the document info so we can detach and reattach later.
             _documentName = documentName;
             _documentCookie = docCookie;
             _windowFrame = windowFrame;
 
             AttachToDocument();
+
+            return true;
         }
 
         /// <summary>
         /// Attaches to the document using cached properties.
         /// </summary>
-        internal void AttachToDocument()
+        private void AttachToDocument()
         {
             ThreadHelper.ThrowIfNotOnUIThread();
-            LineMarker?.AttachToDocument(_documentName, _documentCookie, _windowFrame);
 
-            foreach (LocationModel location in Locations)
-            {
-                location.LineMarker?.AttachToDocument(_documentName, _documentCookie, _windowFrame);
+            IComponentModel componentModel = (IComponentModel)AsyncPackage.GetGlobalService(typeof(SComponentModel));
+            if (componentModel == null)
+            { 
+                return;
             }
 
-            foreach (LocationModel location in RelatedLocations)
-            {
-                location.LineMarker?.AttachToDocument(_documentName, _documentCookie, _windowFrame);
-            }
+            ISarifLocationProviderFactory sarifLocationProviderFactory = componentModel.GetService<ISarifLocationProviderFactory>();
```

Comment:
var as you can get the type from the right.

Reply:
their coding style is to only use "var" when using new.

---
In reply to: [479670488](https://github.com/microsoft/sarif-visualstudio-extension/pull/186#discussion_r479670488) [](ancestors = 479670488)

Reply:
FYI we also use `var` with casts and with `as`:

```
var x = (T)y;
var z = q as T;
```

IOW when and only when the type is explicit on the rhs.

---
In reply to: [480234580](https://github.com/microsoft/sarif-visualstudio-extension/pull/186#discussion_r480234580) [](ancestors = 480234580,479670488)